packages/experimental/RadioGroup/src/Radio/useRadio.ts

```diff
@@ -0,0 +1,97 @@
+import { RadioProps, RadioState } from './Radio.types';
+import * as React from 'react';
+import { useRadioGroupContext } from '../RadioGroup/radioGroupContext';
+import { useAsPressable, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+
+const defaultAccessibilityActions = [{ name: 'Select' }];
+
+export const useRadio = (props: RadioProps): RadioState => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    label,
+    value,
+    disabled,
+    accessibilityActions,
+    accessibilityLabel,
+    componentRef = defaultComponentRef,
+    accessibilityPositionInSet,
+    accessibilitySetSize,
+    enableFocusRing,
+    ...rest
+  } = props;
+
+  // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+  const selectedInfo = useRadioGroupContext();
+
+  const buttonRef = useViewCommandFocus(componentRef);
+
+  /* We don't want to call the user's onChange multiple times on the same selection. */
+  const changeSelection = React.useCallback(() => {
+    if (value != selectedInfo.value) {
+      selectedInfo.onChange && selectedInfo.onChange(value);
+      selectedInfo.updateSelectedButtonRef && componentRef && selectedInfo.updateSelectedButtonRef(componentRef);
+    }
+  }, [selectedInfo, value, componentRef]);
+
+  /* We use the componentRef of the currently selected button to maintain the default tabbable
+    element in a RadioGroup. Since the componentRef isn't generated until after initial render,
+    we must update it once here. */
+  React.useEffect(() => {
+    if (value == selectedInfo.value) {
+      selectedInfo.updateSelectedButtonRef && componentRef && selectedInfo.updateSelectedButtonRef(componentRef);
+    }
+  }, []);
+
+  // Ensure focus is placed on button after click
+  const changeSelectionWithFocus = useOnPressWithFocus(componentRef, changeSelection);
+
+  /* RadioButton changes selection when focus is moved between each RadioButton and on a click */
+  const pressable = useAsPressable({
+    ...rest,
+    onPress: changeSelectionWithFocus,
+    onFocus: changeSelection,
+  });
+
+  const accessibilityActionsProp = React.useCallback(() => {
```

Comment:
post-merge commit:

This shouldn't be a callback function since it's not a function you're calling as a result of an event. It's also currently set to not change after first render, so if accessibilityActions change, as an example, that won't be recognized by the component.

You should use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) for this code.